File: examples/typescript-vitest/steps/roman-numerals.steps.ts
import { steps } from '@varar/varar'
import { toRoman } from '../src/roman-numerals'

const { sensor } = steps()

sensor('a decimal and a roman number', (_state, row: { decimal: string; roman: string }) => {
  return { decimal: row.decimal, roman: toRoman(Number(row.decimal)) }
})

File: examples/java-junit-gradle/src/test/java/examples/RomanNumeralsSteps.java
package examples;

import dev.varar.State;
import dev.varar.StepDefinitions;
import dev.varar.Steps;
import java.util.Map;

public final class RomanNumeralsSteps implements StepDefinitions<RomanNumeralsSteps.Ctx> {

    record Ctx() implements State {}

    @Override
    public void register(Steps<Ctx> s) {
        s.defineState(Ctx::new);

        s.sensor(
                "a decimal and a roman number",
                (Ctx ctx, Map<String, String> row) -> Map.of(
                        "decimal", row.get("decimal"),
                        "roman", RomanNumerals.toRoman(Integer.parseInt(row.get("decimal")))));
    }
}
